tests/unit/models/primo.js

```diff
@@ -617,7 +618,120 @@ describe("Primo Model >", function() {
     });
   });
 
-  describe("primo model buildTemplate method >", function() {
+  describe("primo model isDefaultCoverImage method >", function() {
+    it("should return false when an actual coverImageUrl is returned by the API", function() {
+      var coverImageUrl = "https://assets.thirdiron.com/images/covers/0028-4793.png";
+      expect(primo.isDefaultCoverImage(coverImageUrl)).toEqual(false);
+    });
+
+    it("should return true when a default coverImageUrl is returned by the API", function() {
+      var coverImageUrl = "https://assets.thirdiron.com/images/covers/default.png";
+      expect(primo.isDefaultCoverImage(coverImageUrl)).toEqual(true);
+    });
+  });
+
+  describe("primo model getDirectToPDFUrl method >", function() {
+    it("should not include a fullTextFile property in the BrowZine API response for a journal", function() {
+      var scope = {
+        result: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              issn: ["0096-6762", "0028-4793"]
+            }
+          }
+        }
+      };
+
+      var data = primo.getData(journalResponse);
+
+      expect(data).toBeDefined();
+
+      expect(primo.getDirectToPDFUrl(scope, data)).toBeNull();
+    });
+
+    it("should include a fullTextFile property in the BrowZine API response for an article", function() {
+      var scope = {
+        result: {
+          pnx: {
+            display: {
+              type: ["article"]
+            },
+
+            addata: {
+              issn: ["0028-4793"],
+              doi: ["10.1136/bmj.h2575"]
+            }
+          }
+        }
+      };
+
+      var data = primo.getData(articleResponse);
+
+      expect(data).toBeDefined();
+
+      expect(primo.getDirectToPDFUrl(scope, data)).toEqual("https://develop.browzine.com/libraries/XXX/articles/55134408/full-text-file");
+    });
+  });
+
+  describe("primo model showDirectToPDFLink method >", function() {
+    afterEach(function() {
+      delete browzine.primoArticlePDFDownloadLinkEnabled;
+    });
+
+    it("should show direct to pdf link when configuration property is undefined or null", function() {
+      expect(primo.showDirectToPDFLink()).toEqual(true);
+    });
+
+    it("should show direct to pdf link when configuration property is true", function() {
+      browzine.primoArticlePDFDownloadLinkEnabled = true;
+      expect(primo.showDirectToPDFLink()).toEqual(true);
+    });
+
+    it("should hide direct to pdf link when configuration property is false", function() {
```

Comment:
👍 to this unit test.

Do we want an acceptance-level test that checks to ensure there's no element that appears when this configuration property is false, too?  Similar to [this assertion](https://github.com/thirdiron/browzine-discovery-service-adapters/pull/45/files#diff-e918fc280a5c48d536bacb6a76a7de33R730), but we'd have a test and setup that makes sure the fully-output data truly excludes the PDF stuff.  

Basically, a test to ensure that `showDirectToPDFLink()` has an impact on the final HTML.

Reply:
Good idea Karl. Adding these now.